internal/DebounceDecorator/DebounceDecorator.js

```diff
@@ -60,66 +60,66 @@ const defaultConfig = {
 const DebounceDecorator = hoc(defaultConfig, (config, Wrapped) => {
 	const {cancel, debounce, delay} = config;
 
-	return class extends Component {
-		static displayName = 'DebounceDecorator';
-
-		static propTypes = /** @lends sandstone/internal/DebounceDecorator.DebounceDecorator.prototype */ {
-			/**
-			 * Handler for `onChange` events
-			 *
-			 * `'onChange'` can be changed to a different prop name by specifying the `debounce`
-			 * config option.
-			 *
-			 * @see {@link sandstone/internal/DebounceDecorator.DebounceDecorator.defaultConfig#debounce}
-			 * @name onChange
-			 * @memberof sandstone/internal/DebounceDecorator.DebounceDecorator.prototype
-			 * @type {Function}
-			 * @public
-			 */
-			[debounce]: PropTypes.func
-		};
-
-		constructor (props) {
-			super(props);
-			this.job = new Job(this.emitEvent.bind(this), delay);
-		}
-
-		componentWillUnmount () {
-			this.job.stop();
-		}
+	const Debounce = (props) => {
```

Comment:
```suggestion
	// eslint-disable-next-line no-shadow
	const DebounceDecorator = (props) => {
```
We usually named a returned component as the same as the returning component for higher order components with eslint disabling rule for no-shadow. Not mandatory but if you don't mind... :)